crypto/transactions/base.py

```diff
@@ -0,0 +1,113 @@
+from binascii import hexlify
+from hashlib import sha256
+from struct import pack
+
+from base58 import b58decode_check
+
+from binary.hex.writer import write_high, write_low
+from binary.unsigned_integer.writer import write_bit32, write_bit64
+
+from crypto.identity.keys import public_key_from_secret
+from crypto.message import sign_message, verify_message
+
+
+class BaseTransaction(object):
+
+    def __init__(self):
```

Comment:
Would be good to initialize `self.transaction_type = None` or raise an error in `get_type` function if it's not set.

Reply:
as discussed via DM, this can't be done because `transaction_type` is not available at this point